File: Python/So_hoan_hao.py
def So_hoan_hao(n):
    tong = 0;
    for i in range(1,n):
        if(n%i==0):
            tong = tong + i;
    if(tong==n):
        print("YES");
    else:
        print("NO");
# main
# n = int(input());
So_hoan_hao(9550336);

File: C_C++/So_hoan_hao.c
#include <stdio.h>

void so_hoan_hao()
{
	int n;
	scanf("%d",&n);
	int tong = 0;
	for(int i=1;i<n;i++)
		if(n%i==0)
			tong = tong + i;
	if(tong==n)
		printf("YES");
	else
		printf("NO");
}
int main()
{
	so_hoan_hao();
}